python/metatensor-torch/metatensor/torch/atomistic/io.py

```diff
@@ -0,0 +1,148 @@
+import io
+import os
+from pathlib import Path
+from typing import Union
+
+import numpy as np
+
+import torch
+from metatensor.torch import load_block_buffer
+from metatensor.torch import save_buffer as metatensor_torch_save_buffer
+
+from . import (
+    NeighborListOptions,
+    System,
+)
+
+
+def save(path: Union[str, Path], data: System) -> None:
+    """Save a System object to a file.
+
+    The provided System must contain float64 data and be on the CPU device.
+    Neighbor list requestors will not be saved.
+
+    :param path: The path to save the System object to.
+    :param data: The System object to save.
+    """
+
+    if not isinstance(data, torch.ScriptObject):
+        raise ValueError("`data` must be a scripted object.")
+
+    if _is_system(data):
+        _save_system(path, data)
+    else:
+        raise ValueError("`data` must be a System object.")
+
+
+def load(path: Union[str, Path]) -> System:
+    """Load a System object from a file.
+
+    The loaded System object will be on the CPU device and contain float64 data.
+
+    :param path: The path to load the System object from.
+    :return: The System object.
+    """
+    if not os.path.exists(path):
+        raise FileNotFoundError(f"File not found: {path}")
+
+    if _is_system_npz(path):
+        return _load_system(path)
+    else:
+        raise ValueError(f"File does not contain a valid System object: {path}")
+
+
+def _is_system_npz(path: Union[str, Path]) -> bool:
+    try:
+        data = np.load(path, allow_pickle=True)
+        data["positions"]
+        data["cell"]
+        data["types"]
+        data["neighbor_lists"]
+        return True
+    except Exception:
+        return False
+
+
+def _load_system(path: Union[str, Path]) -> System:
+    data = np.load(path, allow_pickle=True)
+    positions = torch.from_numpy(data["positions"])
+    cell = torch.from_numpy(data["cell"])
+    types = torch.from_numpy(data["types"])
+    pbc = torch.from_numpy(data["pbc"])
+
+    neighbor_list_dict = np.load(io.BytesIO(data["neighbor_lists"]), allow_pickle=True)
+    neighbor_list_dict = {
+        key: torch.from_numpy(value) for key, value in neighbor_list_dict.items()
+    }
+
+    extra_data_dict = np.load(io.BytesIO(data["extra_data"]), allow_pickle=True)
+    extra_data_dict = {
+        key: torch.from_numpy(value) for key, value in extra_data_dict.items()
+    }
```

Comment:
Pickled dict is not something we will be able to load easily from C++ down the line. I would rather do something using file path, like 

```
neighbors/
                      0/
                               options.json
                               values.npz
                      1/
                               options.json
                               values.npz
                      2/
                               options.json
                               values.npz
```

Where `values.npz` is the TensorBlock serialization

For data we should be able to do 


```
data/
        <name-1>.npz
        <name-2>.npz
        <name-3>.npz
```

This might mean we will have to use the `zipfile` module directly, instead of relying on `np.load`